packages/shared/src/utils/generate-joi.ts

```diff
@@ -104,6 +104,12 @@ export function generateJoi(filter: FieldFilter, options?: JoiOptions): AnySchem
 		const getNumberSchema = () => (schema[key] ?? Joi.number()) as NumberSchema;
 		const getDateSchema = () => (schema[key] ?? Joi.date()) as DateSchema;
 
+		const preventEmptyString = (val: any) => {
+			if (typeof val === 'string') return val;
+			// Prevent errors in string functions from Joi and prevent coercing to an empty string
+			return val === null || val === undefined ? '==directus_missing_conditional_string==' : val.toString();
```

Comment:
Is there any way we can prevent having to rely on a hardcoded string like this? Even though it's probably unique enough, I'm also a little worried about falling back to an actual value that will still have the possibility to match 🤔

Reply:
@rijkvanzanten Can we force a `false` of the rule by setting the schema as `Joi.any().equal(true)` for string required filters but missing?

Reply:
There must be a way, but I don't know! Worth some research though 🙂

Reply:
Updated, no longer use a hardcoded string. The generated Joi schema using "string" type filters such as `_contains` or `_starts_with` with null values will always result in false.

Rationale: `null` or `undefined` is not equal to `true`, thus it'll be false.